src/main/java/com/mraof/minestuck/client/gui/DialogueScreen.java

```diff
@@ -0,0 +1,192 @@
+import com.mraof.minestuck.network.DialogueTriggerPacket;
+import com.mraof.minestuck.network.MSPacketHandler;
+import com.mraof.minestuck.entity.dialogue.Dialogue;
+import com.mraof.minestuck.util.DialogueManager;
+import com.mraof.minestuck.entity.dialogue.Trigger;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.Tooltip;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraftforge.client.gui.widget.ExtendedButton;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Displays a screen when interacting with any dialogue capable entity that has a valid dialogue.
+ * It is configurable in terms of what the gui image is, what sort of animated talk sprite shows up, and how many response options there are.
+ */
+public class DialogueScreen extends Screen
+{
+	private static final int GUI_WIDTH = 224;
+	private static final int GUI_HEIGHT = 176;
+	
+	private final ResourceLocation guiBackground;
+	
+	private final LivingEntity entity;
+	private final Dialogue dialogue;
+	private final CompoundTag conditionChecks;
+	
+	private int xOffset;
+	private int yOffset;
+	
+	private final List<Button> responseButtons = new ArrayList<>();
+	
+	DialogueScreen(LivingEntity entity, Dialogue dialogue, CompoundTag conditionChecks)
+	{
+		super(Component.empty());
+		
+		this.entity = entity;
+		this.guiBackground = dialogue.guiPath();
+		this.dialogue = dialogue;
+		this.conditionChecks = conditionChecks;
+	}
+	
+	@Override
+	public void init()
+	{
+		//TODO static gui height/width may not make sense with customizable gui sizing
+		yOffset = (this.height / 2) - (GUI_HEIGHT / 2);
+		xOffset = (this.width / 2) - (GUI_WIDTH / 2);
+		
+		recreateResponseButtons();
+	}
+	
+	public void recreateResponseButtons()
+	{
+		responseButtons.forEach(this::removeWidget);
+		responseButtons.clear();
+		
+		List<Dialogue.Response> filteredResponses = new ArrayList<>();
+		
+		//removes responses if they fail their conditions and should be hidden when that happens
+		for(Dialogue.Response response : dialogue.responses())
+		{
+			if(responseFailedCheck(response.response()) && response.hideIfFailed())
+				continue;
+			
+			filteredResponses.add(response);
+		}
+		
+		for(int i = 0; i < filteredResponses.size(); i++)
+		{
+			Dialogue.Response response = filteredResponses.get(i);
+			String responseMessage = response.response();
+			int yPositionOffset = 20 * i;
+			
+			Component buttonComponent = Component.translatable(responseMessage);
+			
+			ExtendedButton entryButton = new ExtendedButton(xOffset + 20, yOffset + 40 + yPositionOffset, 190, 14, buttonComponent,
+					button -> clickResponse(responseMessage));
```

Comment:
Assuming that `this.dialogue` or `this.dialogue.responses()` won't change as long as this screen exists, you could pass `response` directly into `clickResponse()` without going the roundabout way through `responseMessage`.